src/Servers/Kestrel/Transport.Sockets/src/SocketTransport.cs

```diff
@@ -65,8 +63,9 @@ internal sealed class SocketTransport : ITransport
             }
             else
             {
-                _numSchedulers = ThreadPoolSchedulerArray.Length;
-                _schedulers = ThreadPoolSchedulerArray;
+                var directScheduler = new PipeScheduler[] { PipeScheduler.ThreadPool };
+                _numSchedulers = directScheduler.Length;
+                _schedulers = directScheduler;
```

Comment:
Not that I'm arguing this static array was particularly helpful, why remove it?

Reply:
Its only used if ioQueueCount is manually set to 0 at start up; and then only used one to assign here, so why initialise it in a readonly static? 

It suggests an entirely different lifetime.

Reply:
I don't think it's that confusing. We use private statics as pseudo-constants all over the place in non-static Kestrel classes.

That being said, tne extra allocation per connection in a non-default config certainly doesn't bother me. I just making sure there wasn't some deeper reason for removing it.

Reply:
Not greatly :)